### PR TITLE
Fix data race reported on SetIsInSecondaryCache in LRUCache

### PR DESCRIPTION
**Pull request by gitbw95**
Summary:
Currently, `SetIsInSecondaryCache` is after `Promote`. After `Promote`, a handle can be accessed and its flags can be set. This causes data race.

Test Plan:
unit tests
stress tests

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -20,6 +20,7 @@
 * Fix data race bug in hash linked list memtable. With this bug, read request might temporarily miss an old record in the memtable in a race condition to the hash bucket.
 * Fix a bug that `best_efforts_recovery` may fail to open the db with mmap read.
 * Fixed a bug where blobs read during compaction would pollute the cache.
+* Fixed a bug where a data race can happens for setting flags (e.g. `SetIsInSecondaryCache()`) after a handle from secondary cache is `Promote()`.
```
Comment (pdillinger) [inline HISTORY.md:23]: HISTORY is intended for API users. This text is about implementation details. What is the user perspective on this bug fix? Who is potentially affected? (only secondary cache users, I believe)

**Review by pdillinger — APPROVED** [review]
Otherwise LGTM!

File: HISTORY.md (hunk 1 above)
Comment (gitbw95) [reply HISTORY.md]: Update HISTORY.md accordingly.

File: HISTORY.md [hunk 2]
```diff
@@ -20,6 +20,7 @@
 * Fix data race bug in hash linked list memtable. With this bug, read request might temporarily miss an old record in the memtable in a race condition to the hash bucket.
 * Fix a bug that `best_efforts_recovery` may fail to open the db with mmap read.
 * Fixed a bug where blobs read during compaction would pollute the cache.
+* Fixed a bug where a data race can happens when a handle from secondary cache is `Promote()` and then both the `SetIsInSecondaryCache()` and a new `Lookup()` try to set flags.
```
Comment (pdillinger) [inline HISTORY.md:23]: Still refers to implementation details. I would say, "Fixed a data race in LRUCache when used with a secondary_cache." (If someone wants more detail, all they have to do is `git blame HISTORY.md`, find the commit associated with this release note, and look at the details.)

Comment (gitbw95) [reply HISTORY.md]: The comment is addressed in the latest commit.

**Pull request closed without merging**